prov/gni/configure.m4

```diff
@@ -0,0 +1,160 @@
+dnl Configury specific to the libfabrics GNI provider
+
+dnl Called to configure this provider
+dnl
+dnl Arguments:
+dnl
+dnl $1: action if configured successfully
+dnl $2: action if not configured successfully
+dnl
+
+# _PKG_CONFIG2([VARIABLE], [COMMAND1], [COMMAND2], [MODULES])
+# ---------------------------------------------
+m4_define([_PKG_CONFIG2],
+[if test -n "$$1"; then
+	pkg_cv_[]$1="$$1"
+elif test -n "$PKG_CONFIG"; then
+	PKG_CHECK_EXISTS([$4],
+		[pkg_cv_[]$1=`$PKG_CONFIG --[]$2 --[]$3 "$4" 2>/dev/null`
+		test "x$?" != "x0" && pkg_failed=yes ],
+		[pkg_failed=yes])
+else
+	pkg_failed=untried
+fi[]dnl
+])# _PKG_CONFIG2
+
+# PKG_CHECK_MODULES_STATIC(VARIABLE-PREFIX, MODULES, [ACTION-IF-FOUND],
+# [ACTION-IF-NOT-FOUND])
+#
+#
+# Variation of PGK_CHECK_MODULES which also defines $1_LIBS_STATIC
+# by using the pkg-config --libs --static pkg-config option.
+#
+# --------------------------------------------------------------
+AC_DEFUN([PKG_CHECK_MODULES_STATIC],
```

Comment:
Meta questions: 

1. Should these PKG_ and _PKG macros be up in config/fi_pkg.m4?  I.e., could other providers use them?  I know that config/pkg.m4 exists, but I'm assuming that that file is directly downloaded from elsewhere, and these macros that you have here are libfabric-specific (they don't appear to be GNI specific...?).
1. Shouldn't these macros *not* start with PKG (or _PKG) so as to differentiate them from the macros in config/pkg.m4 (i.e., the macros that come from pkg-config upstream)?